src/files/XmlFile.ts

```diff
@@ -454,16 +454,20 @@ export class XmlFile {
             return script;
         });
 
-        const extraImportIdx = this.ast.component.scripts.length - 1;
```

Comment:
Rather than mess with array indexes...now we just push a new array, and restore the old one after the fact.